subprojects/kotlinTest/src/test/kotlin/org/mockito/kotlin/SuspendTest.kt

```diff
@@ -0,0 +1,107 @@
+package org.mockito.kotlin
+
+import kotlinx.coroutines.experimental.runBlocking
+import org.hamcrest.core.IsEqual
+import org.junit.Assert.assertThat
+import org.junit.Test
+import org.mockito.Mockito.*
+
+/**
+ * Test for mocking and verifying Kotlin suspending functions.
+ */
+class SuspendTest {
+    interface SuspendableInterface {
+        suspend fun suspendFunction(x: Int): Int
+        suspend fun suspendFunctionWithDefault(x: Int = 1): Int
+    }
+
+    open class SuspendableClass : SuspendableInterface {
+        suspend override fun suspendFunction(x: Int): Int = error("not implemented")
+        suspend override fun suspendFunctionWithDefault(x: Int): Int = error("not implemented")
+        suspend open fun suspendClassFunctionWithDefault(x: Int = 1): Int = error("not implemented")
+    }
+
+    open class CallsSuspendable(private val suspendable: SuspendableInterface) {
+        fun callSuspendable(x: Int) = runBlocking {
+            suspendable.suspendFunction(x)
+        }
+
+        fun suspendFunctionWithDefault() = runBlocking {
+            suspendable.suspendFunctionWithDefault()
+        }
+
+        fun suspendFunctionWithDefault(x: Int) = runBlocking {
+            suspendable.suspendFunctionWithDefault(x)
+        }
+    }
+
+    class CallsSuspendableClass(private val suspendable: SuspendableClass) : CallsSuspendable(suspendable) {
+        fun suspendClassFunctionWithDefault() = runBlocking {
+            suspendable.suspendClassFunctionWithDefault()
+        }
+
+        fun suspendClassFunctionWithDefault(x: Int) = runBlocking {
+            suspendable.suspendClassFunctionWithDefault(x)
+        }
+    }
+
+    @Test
+    fun shouldMockSuspendInterfaceFunction() = runBlocking<Unit> {
+        val mockSuspendable: SuspendableInterface = mock(SuspendableInterface::class.java)
+        val callsSuspendable = CallsSuspendable(mockSuspendable)
+
+        `when`(mockSuspendable.suspendFunction(1)).thenReturn(2)
+        `when`(mockSuspendable.suspendFunction(intThat { it >= 2 })).thenReturn(4)
+
+        assertThat(callsSuspendable.callSuspendable(1), IsEqual(2))
+        assertThat(callsSuspendable.callSuspendable(2), IsEqual(4))
+
+        verify(mockSuspendable).suspendFunction(1)
+        verify(mockSuspendable).suspendFunction(2)
+    }
+
+    @Test
+    fun shouldMockSuspendInterfaceFunctionWithDefault() = runBlocking<Unit> {
+        val mockSuspendable: SuspendableInterface = mock(SuspendableInterface::class.java)
+        val callsSuspendable = CallsSuspendable(mockSuspendable)
+
+        `when`(mockSuspendable.suspendFunctionWithDefault()).thenReturn(2)
+        `when`(mockSuspendable.suspendFunctionWithDefault(intThat { it >= 2 })).thenReturn(4)
```

Comment:
@elizarov I'm curious, wouldn't be easier to just use `given/willReturn` instead of quoted `when/thenReturn` in Kotlin code?